internal/models/challenge.go

```diff
@@ -2,21 +2,26 @@ package models
 
 import (
 	"database/sql"
+	"github.com/go-webauthn/webauthn/protocol"
+	"github.com/go-webauthn/webauthn/webauthn"
 	"github.com/gofrs/uuid"
 	"github.com/pkg/errors"
 	"github.com/supabase/auth/internal/storage"
 	"time"
 )
 
 type Challenge struct {
-	ID         uuid.UUID  `json:"challenge_id" db:"id"`
-	FactorID   uuid.UUID  `json:"factor_id" db:"factor_id"`
-	CreatedAt  time.Time  `json:"created_at" db:"created_at"`
-	VerifiedAt *time.Time `json:"verified_at,omitempty" db:"verified_at"`
-	IPAddress  string     `json:"ip_address" db:"ip_address"`
-	Factor     *Factor    `json:"factor,omitempty" belongs_to:"factor"`
+	ID                uuid.UUID  `json:"challenge_id" db:"id"`
+	FactorID          uuid.UUID  `json:"factor_id" db:"factor_id"`
+	CreatedAt         time.Time  `json:"created_at" db:"created_at"`
+	VerifiedAt        *time.Time `json:"verified_at,omitempty" db:"verified_at"`
+	IPAddress         string     `json:"ip_address" db:"ip_address"`
+	Factor            *Factor    `json:"factor,omitempty" belongs_to:"factor"`
+	WebauthnChallenge *string    `json:"webauthn_challenge,omitempty" db:"webauthn_challenge"`
```

Comment:
I wonder why not encode the WebAuthn [SessionData](https://pkg.go.dev/github.com/go-webauthn/webauthn@v0.11.0/webauthn#SessionData) in here?

Reply:
The `challenge` is encoded in it as well and it fits perfectly.

Reply:
That's a good suggestion- I would actually prefer that. I was initially under the impression we didn't want to store fields beyond the challenge.